packages/react-core/src/components/ExpandableSection/ExpandableSection.tsx

```diff
@@ -127,25 +168,21 @@ export class ExpandableSection extends React.Component<ExpandableSectionProps, E
           displaySize === 'large' && styles.modifiers.displayLg,
           isWidthLimited && styles.modifiers.limitWidth,
           isIndented && styles.modifiers.indented,
+          truncateContent && styles.modifiers.truncate,
           className
         )}
+        {...props}
       >
-        {!isDetached && (
-          <button
-            className={css(styles.expandableSectionToggle)}
-            type="button"
-            aria-expanded={propOrStateIsExpanded}
-            onClick={() => onToggle(!propOrStateIsExpanded)}
-          >
-            <span className={css(styles.expandableSectionToggleIcon)}>
-              <AngleRightIcon aria-hidden />
-            </span>
-            <span className={css(styles.expandableSectionToggleText)}>{toggleContent || computedToggleText}</span>
-          </button>
-        )}
-        <div className={css(styles.expandableSectionContent)} hidden={!propOrStateIsExpanded} id={contentId}>
+        {!truncateContent && expandableToggle}
+        <div
+          ref={this.expandableContentRef}
+          className={css(styles.expandableSectionContent)}
+          hidden={!propOrStateIsExpanded}
```

Comment:
I noticed `hidden` is still being applied to the content element, though it's being ignored when the element renders on the page since we have `display: -webkit-box` to enable the CSS line clamp feature. It also seems like VO is reading the content for this variation even with `hidden` - I'm assuming because it detects it's visible? So it doesn't look like it's doing anything - should we maybe remove it and add `aria-hidden="true"` to the toggle since the content should be read like a normal element to a screen reader?

Reply:
For the comments on the detached example, adding that line in to add the truncate class does work, though yeah there is a bit of extra top padding:

![detached example with truncation](https://user-images.githubusercontent.com/70952936/185406532-ff0b51df-c42d-4a45-997d-81ea3e3e6b85.png)

> Also just thinking, but I wonder if the next ask is going to be to detect if there is enough content to truncate. If so, truncate it and show the toggle, otherwise just show the content.

I was thinking about that as well, whether that should be something the component should determine internally or if it would be something for the consumer to have to determine.

> I noticed hidden is still being applied to the content element, though it's being ignored when the element renders on the page since we have display: -webkit-box to enable the CSS line clamp feature. It also seems like VO is reading the content for this variation even with hidden - I'm assuming because it detects it's visible? So it doesn't look like it's doing anything - should we maybe remove it and add aria-hidden="true" to the toggle since the content should be read like a normal element to a screen reader?

I'd agree the `hidden` attribute could be removed when the content is truncated, but we should avoid using aria-hidden in its place. With the layout of the content and toggle, the entire content would end up skipped by assistive tech and, even though the "show more/less" toggle could indicate the collapsed state, users most likely wouldn't know to navigate backwards for that content that is now expanded.

Reply:
> For the comments on the detached example, adding that line in to add the truncate class does work, though yeah there is a bit of extra top padding

Do you want to open a core PR for that or want me to?

Reply:
> I'd agree the hidden attribute could be removed when the content is truncated, but we should avoid using aria-hidden in its place.

Oh sorry, I was saying we would add `aria-hidden` to the toggle since the truncated text is only visually hidden, but visible to screen readers.

Reply:
>Do you want to open a core PR for that or want me to?

I can put one up in a bit!

>Oh sorry, I was saying we would add aria-hidden to the toggle since the truncated text is only visually hidden, but visible to screen readers.

Ah I misread! Thanks for clarifying that. I believe that would work well! I'll let you know if it seems like it won't, otherwise I'll push that change with other changes.